shared/js/ui/base/safari-ui-wrapper.es6.js

```diff
@@ -1,45 +1,55 @@
 /* global safari:false */
+let context
+
+if (safari &&
+        safari.extension &&
+        safari.extension.globalPage &&
+        safari.extension.globalPage.contentWindow) {
+    context = 'popup'
+} else if (safari &&
+        safari.self &&
+        safari.self.tab) {
+    context = 'options'
+} else {
+    throw new Error('safari-ui-wrapper couldn\'t figure out the context it\'s in')
+}
+
+let sendOptionsMessage = (message, resolve, reject) => {
+    if (message.whitelisted) {
+        resolve(safari.self.tab.dispatchMessage('whitelisted', message))
+    } else if (message.getSetting) {
+        // send message random ID so we know which promise to res
+        let id = Math.random()
+        message.id = id
+        safari.self.tab.dispatchMessage('getSetting', message)
+
+        safari.self.addEventListener('message', (e) => {
+            if (e.name === 'getSetting' && e.message.id === id) {
+                delete e.message.id
+                resolve(e.message)
+            }
+        }, true)
```

Comment:
Note, I realized this repeatedly adds event listeners without removing any of them. Probably not a big issue since users will rarely change settings on the options page to the point where it would be a problem. Worth keeping in mind for the future though.